skimage/filters/thresholding.py

```diff
@@ -713,3 +716,283 @@ def threshold_triangle(image, nbins=256):
+    window_size : int, optional
+        Odd size of pixel neighborhood window (e.g. 3, 5, 7,
+        ..., 21, ...).
+    k : float, optional
+        Value of parameter k in threshold formula.
+    offset : float, optional
+        Constant subtracted from obtained local thresholds.
+
+    Returns
+    -------
+    threshold : (N, M) ndarray
+        Local threshold values. All pixels with an intensity higher than
+        this value are assumed to be foreground.
+
+    References
+    ----------
+    .. [1] Niblack, W (1986), An introduction to Digital Image
+           Processing, Prentice-Hall.
+
+    Examples
+    --------
+    >>> from skimage import data
+    >>> image = data.page()
+    >>> binary_image = threshold_niblack(image, window_size=7, k=0.1)
+    """
+    if len(image.shape) != 2:
+        raise ValueError("Image is not 2D grayscale.")
+
+    m, s = _mean_std(image, window_size)
+    t = m - k * s
+    return t - offset
+
+
+def threshold_sauvola(image, window_size=15, k=0.2, r=128., offset=0):
+    """Applies Sauvola local threshold to an array. Sauvola is a
+    modification of Niblack technique.
+
+    In the original method a threshold T is calculated for every pixel
+    in the image using the following formula:
+
+    T = m(x,y) * (1 + k * ((s(x,y) / R) - 1)) - offset
+
+    where m(x,y) and s(x,y) are the mean and standard deviation of
+    pixel (x,y) neighborhood defined by a rectangular window with size w
+    times w centered around the pixel. k is a configurable parameter
+    that weights the effect of standard deviation.
+    R is the maximum standard deviation of a greyscale image (R = 128).
+
+    Parameters
+    ----------
+    image: (N, M) ndarray
+        Input image. Only 2D grayscale images allowed.
+    window_size : int, optional
+        Odd size of pixel neighborhood window (e.g. 3, 5, 7...).
+    k : float, optional
+        Value of parameter k in threshold formula.
+    r : float, optional
+        Value of R in threshold formula. Not used by method 'wolf'.
+    offset : float, optional
+        Constant subtracted from obtained local thresholds.
+
+    Returns
+    -------
+    threshold : (N, M) ndarray
+        Local threshold values. All pixels with an intensity higher than
+        this value are assumed to be foreground.
+
+    References
+    ----------
+    .. [1] J. Sauvola and M. Pietikainen, "Adaptive document image
+           binarization," Pattern Recognition 33(2),
+           pp. 225-236, 2000.
+           DOI:10.1016/S0031-3203(99)00055-2
+
+    Examples
+    --------
+    >>> from skimage import data
+    >>> image = data.page()
+    >>> binary_sauvola = threshold_sauvola(image,
+    ...                                    window_size=15, k=0.2, r=128)
+    """
+    image_norm = None
+    if len(image.shape) != 2:
+        raise ValueError("Image is not 2D grayscale.")
+    m, s = _mean_std(image, window_size)
+    t = m * (1 + k * ((s / r) - 1)) # TODO inconsistant with docstring
```

Comment:
Was just a reminder for myself. There is a mix between r and R, and I found the notations a bit inconsistent (with an e :) )